gprofiler/spark/collector.py

```diff
@@ -372,32 +324,62 @@ def _get_running_apps(self) -> Dict[str, Tuple[str, str]]:
         if self._cluster_mode == SPARK_YARN_MODE:
             running_apps = self._yarn_init()
             return self._get_spark_app_ids(running_apps)
-        elif self._cluster_mode == SPARK_DRIVER_MODE:
-            return self._driver_init()
+        elif self._cluster_mode == SPARK_STANDALONE_MODE:
+            return self._standalone_init()
         elif self._cluster_mode == SPARK_MESOS_MODE:
             return self._mesos_init()
         else:
             raise ValueError(f"Invalid cluster mode {self._cluster_mode!r}")
 
-    def _driver_init(self) -> Dict[str, Tuple[str, str]]:
+    def _standalone_init(self) -> Dict[str, Tuple[str, str]]:
         """
         Return a dictionary of {app_id: (app_name, tracking_url)} for the running Spark applications
         """
-        return self._driver_get_apps(status=YARN_RUNNING_APPLICATION_SPECIFIER)
+        metrics_json = self._rest_request_to_json(
+            self._master_address, SPARK_MASTER_STATE_PATH, SPARK_STANDALONE_SERVICE_CHECK
+        )
+        running_apps = {}
+
+        if metrics_json.get("activeapps") is not None:
+            for app in metrics_json["activeapps"]:
+                try:
+                    app_id = app["id"]
+                    app_name = app["name"]
+
+                    # Parse through the HTML to grab the application driver's link
+                    app_url = self._get_standalone_app_url(app_id)
+                    logger.debug(f"app_url is: {app_url}")
+
+                    if app_id and app_name and app_url:
+                        running_apps[app_id] = (app_name, app_url)
+                        logger.debug(f"Application URL: {app_url}")
+                except KeyError:
+                    logger.exception("Key error was found while iterating applications.")
+                except Exception:
+                    # it's possible for the requests to fail if the job
+                    # completed since we got the list of apps.  Just continue
+                    pass
 
-    def _driver_get_apps(self, *args: Any, **kwargs: Any) -> Dict[str, Tuple[str, str]]:
+        return running_apps
+
+    def _get_standalone_app_url(self, app_id: str) -> Any:
         """
-        Return a dictionary of {app_id: (app_name, tracking_url)} for the Spark applications
+        Return the application URL from the app info page on the Spark master.
+        Due to a bug, we need to parse the HTML manually because we cannot
+        fetch JSON data from HTTP interface.
```

Comment:
Talked f2f. Need to elaborate on this scenario, or mention that we just decide to carry over logic from DD.